docs/blocks-storage/store-gateway.template

```diff
@@ -0,0 +1,158 @@
+
+For each block belonging to a store-gateway shard, the store-gateway loads its `meta.json`, the `deletion-mark.json` and the index-header. Once a block is loaded on the store-gateway, it's ready to be queried by queriers. When the querier queries blocks through a store-gateway, the response will contain the list of actually queried block IDs. If a querier tries to query a block which has not been loaded by a store-gateway, the querier will either retry on a different store-gateway (if blocks replication is enabled) or fail the query.
+
+Blocks can be replicated across multiple store-gateway instances based on a replication factor configured via `-experimental.store-gateway.replication-factor`. The blocks replication is used to protect from query failures caused by some blocks not loaded by any store-gateway instance at a given time like, for example, in the event of a store-gateway failure or while restarting a store-gateway instance (e.g. during a rolling update).
+
+This feature can be enabled via `-experimental.store-gateway.sharding-enabled=true` and requires the backend [hash ring](../architecture.md#the-hash-ring) to be configured via `-experimental.store-gateway.sharding-ring.*` flags (or their respective YAML config options).
+
+### Auto-forget
+
+When a store-gateway instance cleanly shutdowns, it automatically unregisters itself from the ring. However, in the event of a crash or node failure, the instance will not be unregistered from the ring, potentially leaving a spurious entry in the ring forever.
+
+To protect from this, when an healthy store-gateway instance finds another instance in the ring which is unhealthy since more then 10 times the configured `-experimental.store-gateway.sharding-ring.heartbeat-timeout`, the healthy instance forcely removes the unhealthy one from the ring.
+
+This feature is called **auto-forget** and is built into the store-gateway.
+
+## Caching
+
+The store-gateway supports the following caches:
+
+- [Index cache](#index-cache)
+- [Chunks cache](#chunks-cache)
+- [Metadata cache](#metadata-cache)
+
+Caching is optional, but **highly recommended** in a production environment.
+
+### Index cache
+
+The store-gateway can use a cache to speed up lookups of postings and series from TSDB blocks indexes. Two backends are supported:
+
+- `inmemory`
+- `memcached`
+
+#### In-memory index cache
+
+The `inmemory` index cache is **enabled by default** and its max size can be configured through the flag `-experimental.tsdb.bucket-store.index-cache.inmemory.max-size-bytes` (or config file). The trade-off of using the in-memory index cache is:
+
+- Pros: zero latency
+- Cons: increased store-gateway memory usage, not shared across multiple store-gateway replicas (when sharding is disabled or replication factor > 1)
+
+#### Memcached index cache
```

Comment:
I guess you mean the metadata cache, but yes, it's duplicated. I'm a bit dubious about moving it to its own document at this stage because I think it's easier to consume if, given a component (eg. store-gateway), you read its doc in a single page. Maybe we can live with this duplicated section for a while?

Reply:
OK, let's keep it for now.